extensions/azurePublishNew/src/components/provisioningWizards/steps/ResourceConfigurationStep.tsx

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+import styled from '@emotion/styled';
+import { useRecoilState } from 'recoil';
+import formatMessage from 'format-message';
+import { FluentTheme, NeutralColors } from '@uifabric/fluent-theme';
+import {
+  ScrollablePane,
+  ScrollbarVisibility,
+  TooltipHost,
+  Icon,
+  Stack,
+  Text,
+  FontWeights,
+  FontSizes,
+  Label,
+  IStackTokens,
+  IStackItemStyles,
+  Link,
+} from 'office-ui-fabric-react';
+
+import {
+  userInfoState,
+  resourceGroupSelectionState,
+  subscriptionSelectionState,
+  tenantSelectionState,
+  deployLocationSelectionState,
+} from '../../../recoilModel/atoms/resourceConfigurationState';
+import { ResourceGroupPicker } from '../../resourceConfiguration/ResourceGroupPicker';
+import { SubscriptionPicker } from '../../resourceConfiguration/SubscriptionPicker';
+import { TenantPicker } from '../../resourceConfiguration/TenantPicker';
+import { DeployLocationPicker } from '../../resourceConfiguration/DeployLocationPicker';
+
+type Props = {
+  onResourceConfigurationChange: (isValidConfiguration: boolean) => void;
+};
+
+const ConfigureResourcesSectionName = styled(Text)`
+  font-size: ${FluentTheme.fonts.mediumPlus.fontSize};
+  font-weight: ${FontWeights.semibold};
+  margin-bottom: 4px;
+`;
+
+const ConfigureResourcesSectionDescription = styled(Text)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  line-height: ${FontSizes.size14};
+  margin-bottom: 20px;
+`;
+
+const configureResourcePropertyStackTokens: IStackTokens = { childrenGap: 5 };
+
+const configureResourcePropertyLabelStackStyles: IStackItemStyles = {
+  root: {
+    width: '200px',
+  },
+};
+
+const ConfigureResourcesPropertyLabel = styled(Label)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  font-weight: ${FontWeights.regular};
+`;
+
+const autoCompleteTextFieldStyles = { root: { paddingBottom: '4px', width: '300px' } };
+
+const configureResourcesIconStyle = {
+  root: {
+    color: NeutralColors.gray160,
+    userSelect: 'none',
+  },
+};
+
+const LearnMoreLink = styled(Link)`
+  user-select: none;
+  font-size: 14px;
+`;
+
+const FullWidthForm = styled.form`
+  width: '100%';
+`;
+
+//TODO: a wrapper function would be nice eg: formatMessages({})
+const messages = {
```

Comment:
as we discussed before, this wont work, need to be invoked within the context of a react component